internal/content_coding.go

```diff
@@ -137,12 +142,13 @@ type GzipEncoder struct {
 	buf    *bytes.Buffer
 }
 
-func NewGzipEncoder() *GzipEncoder {
+func NewGzipEncoder(level int) (*GzipEncoder, error) {
 	var buf bytes.Buffer
+	w, err := gzip.NewWriterLevel(&buf, level)
```

Comment:
I would rather return a `nil` encoder to force the caller to check the error code
```suggestion
	w, err := gzip.NewWriterLevel(&buf, level)
	if err != nil {
		return nil, err
	}
```
You can then return `nil` for the error below.

Same for all other instances here.